src/class-wpml-elementor-media-translation.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class WPML_Elementor_Media_Translation {
+
+	private $data_settings;
+	private $image_translate;
+	private $translation_element_factory;
+
+	public function __construct(
+		IWPML_Page_Builders_Data_Settings $data_settings,
+		WPML_Media_Image_Translate $image_translate,
+		WPML_Translation_Element_Factory $translation_element_factory
+	) {
+		$this->data_settings               = $data_settings;
+		$this->image_translate             = $image_translate;
+		$this->translation_element_factory = $translation_element_factory;
+	}
+
+	public function add_hooks() {
+		add_action( 'wpml_media_after_translate_media_in_post_content', array( $this, 'translate_image' ), PHP_INT_MAX, 3 );
+	}
+
+	public function translate_image( $post_id, $attachment_id, $target_language ) {
+		$post        = $this->translation_element_factory->create( $post_id, 'post' );
+		$translation = $post->get_translation( $target_language );
+
+		$attachment             = $this->translation_element_factory->create( $attachment_id, 'post' );
+		$attachment_translation = $attachment->get_translation( $target_language );
```

Comment:
@adrianosferreira You can move these two lines into the if statement because they are only needed then.